central/sensor/service/pipeline/nodeinventory/two_pipelines_test.go

```diff
@@ -0,0 +1,404 @@
+	imageStoreMock "github.com/stackrox/rox/central/image/datastore/mocks"
+	nodeDatastoreMocks "github.com/stackrox/rox/central/node/datastore/mocks"
+	"github.com/stackrox/rox/central/ranking"
+	riskStoreMock "github.com/stackrox/rox/central/risk/datastore/mocks"
+	"github.com/stackrox/rox/central/risk/manager"
+	"github.com/stackrox/rox/central/sensor/service/pipeline"
+	"github.com/stackrox/rox/central/sensor/service/pipeline/nodes"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/metrics"
+	nodeEnricher "github.com/stackrox/rox/pkg/nodes/enricher"
+	"github.com/stackrox/rox/pkg/scancomponent"
+	"github.com/stackrox/rox/pkg/scanners"
+	"github.com/stackrox/rox/pkg/scanners/types"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+	"golang.org/x/sync/semaphore"
+)
+
+const (
+	nodeID              = "node-id-1"
+	nodeName            = "node-name"
+	clusterID           = "cluster1"
+	kernelComponentName = "kernel"
+)
+
+func Test_TwoPipelines_Run(t *testing.T) {
+	nodeWithScore := &storage.Node{
+		Id:            nodeID,
+		Name:          nodeName,
+		ClusterId:     clusterID,
+		ClusterName:   clusterID,
+		KernelVersion: "v1",
+		Notes:         []storage.Node_Note{storage.Node_MISSING_SCAN_DATA},
+		RiskScore:     1,
+	}
+
+	nodeWithScanWithKernelV1 := &storage.Node{
+		Id:            nodeID,
+		Name:          nodeName,
+		ClusterId:     clusterID,
+		ClusterName:   clusterID,
+		KernelVersion: "v1",
+		Notes:         []storage.Node_Note{},
+		RiskScore:     1,
+		Scan:          nodeScanFixtureWithKernel("v1"),
+		SetComponents: &storage.Node_Components{Components: 1},
+		SetCves:       &storage.Node_Cves{Cves: 1},
+		SetTopCvss:    &storage.Node_TopCvss{TopCvss: 1},
+	}
+
+	nodeWithScanWithKernelV2 := &storage.Node{
+		Id:            nodeID,
+		Name:          nodeName,
+		ClusterId:     clusterID,
+		ClusterName:   clusterID,
+		KernelVersion: "v1",
+		Notes:         []storage.Node_Note{},
+		RiskScore:     1,
+		Scan:          nodeScanFixtureWithKernel("v2"),
+		SetComponents: &storage.Node_Components{Components: 1},
+		SetCves:       &storage.Node_Cves{Cves: 1},
+		SetTopCvss:    &storage.Node_TopCvss{TopCvss: 1},
+	}
+
+	type usedMocks struct {
+		clusterStore      *clusterDatastoreMocks.MockDataStore
+		nodeDatastore     *nodeDatastoreMocks.MockDataStore
+		cveDatastore      *nodeCVEDataStoreMocks.MockDataStore
+		deploymentStorage datastore.DataStore
+		imageStorage      imageDS.DataStore
+		riskStorage       *riskStoreMock.MockDataStore
+		updater           updater.Updater
+	}
+	tests := []struct {
+		name                      string
+		mocks                     usedMocks
+		riskManager               manager.Manager
+		enricher                  nodeEnricher.NodeEnricher
+		operations                []func(t *testing.T, np pipeline.Fragment, ninvp pipeline.Fragment) error
+		wantErr                   string
```

Comment:
There are no test cases that use `wantErr`. Any chance to add tests for error cases? Otherwise best to remove it.

Reply:
Right, I will remove it from here. There are other tests for testing a single pipeline and those seem to be a better place to test the errors.